fence/resources/audit/utils.py

```
@@ -100,7 +100,8 @@ def create_log_for_request(request: Request):
     record a log entry.
     """
     claims = validate_jwt()
-    username = get_user_from_claims(claims).username
+    username = get_user_from_claims(claims).username if "sub" in claims else "None"
+    client_id = claims.get("azp", "None") or "None"
```

should we raise an error if both username and client end up being None?

---

i don't think so, this is just the logging, the actual functional code would catch this (arborist, etc)